src/ServiceControl/Monitoring/HeartbeatMonitoringHostedService.cs

```diff
@@ -24,17 +24,7 @@ public async Task StartAsync(CancellationToken cancellationToken)
             timer = scheduler.Schedule(_ => CheckEndpoints(), TimeSpan.Zero, TimeSpan.FromSeconds(5), e => { log.Error("Exception occurred when monitoring endpoint instances", e); });
         }
 
-        public async Task StopAsync(CancellationToken cancellationToken)
-        {
-            try
-            {
-                await timer.Stop();
-            }
-            catch (OperationCanceledException) when (cancellationToken.IsCancellationRequested)
-            {
-                //NOOP, invoked Stop does not
-            }
-        }
+        public Task StopAsync(CancellationToken cancellationToken) => timer.Stop(cancellationToken);
```

Comment:
Isn't this changing the previous behavior? If this throws an OperationCanceledException and cancellationToken.IsCancellationRequested is true, the exception will bubble up instead of being suppressed like before. Was that intentional?

Reply:
You are right but that now happens in the async timer so end result should be the same? (see https://github.com/Particular/ServiceControl/pull/4891/files#diff-96d7e0170b214f88063570076afc55fe8451f84549113c204650a38ea9cec8c9R88)